crates/ika-core/src/dwallet_mpc/crytographic_computation/mpc_computations/presign.rs

```diff
@@ -4,66 +4,327 @@
+                    dwallet_public_output,
+                )?)
+            }
+            DWalletSignatureScheme::SchnorrkelSubstrate => {
+                let protocol_public_parameters = versioned_network_encryption_key_public_data
+                    .ristretto_protocol_public_parameters()?;
+                PresignPublicInputByProtocol::SchnorrkelSubstrate(generate_presign_public_input::<
+                    RistrettoSchnorrkelSubstrateProtocol,
+                >(
+                    session_identifier,
+                    protocol_public_parameters,
+                    dwallet_public_output,
+                )?)
+            }
+            DWalletSignatureScheme::EdDSA => {
+                let protocol_public_parameters = versioned_network_encryption_key_public_data
+                    .curve25519_protocol_public_parameters()?;
+                PresignPublicInputByProtocol::EdDSA(generate_presign_public_input::<
+                    Curve25519EdDSAProtocol,
+                >(
+                    session_identifier,
+                    protocol_public_parameters,
+                    dwallet_public_output,
+                )?)
+            }
+            DWalletSignatureScheme::ECDSASecp256r1 => {
+                let protocol_public_parameters = versioned_network_encryption_key_public_data
+                    .secp256r1_protocol_public_parameters()?;
+                PresignPublicInputByProtocol::Secp256r1ECDSA(generate_presign_public_input::<
+                    Secp256R1ECDSAProtocol,
+                >(
+                    session_identifier,
+                    protocol_public_parameters,
+                    dwallet_public_output,
+                )?)
+            }
+            DWalletSignatureScheme::Taproot => {
+                let protocol_public_parameters = versioned_network_encryption_key_public_data
+                    .secp256k1_protocol_public_parameters();
+                PresignPublicInputByProtocol::Taproot(generate_presign_public_input::<
+                    Secp256K1TaprootProtocol,
+                >(
+                    session_identifier,
+                    protocol_public_parameters,
+                    dwallet_public_output,
+                )?)
+            }
         };
 
-        Ok(pub_input)
+        Ok(input)
+    }
+}
+
+fn generate_presign_public_input<P: Protocol>(
+    session_identifier: SessionIdentifier,
+    protocol_public_parameters: P::ProtocolPublicParameters,
+    dwallet_public_output: SerializedWrappedMPCPublicOutput,
+) -> DwalletMPCResult<<PresignParty<P> as mpc::Party>::PublicInput> {
+    let VersionedDwalletDKGSecondRoundPublicOutput::V2(dkg_output) =
+        bcs::from_bytes(dwallet_public_output.as_slice())?
+    else {
+        return Err(DwalletMPCError::MPCSessionError {
+                session_identifier,
+                error: "presign public input v2 only supports VersionedDwalletDKGSecondRoundPublicOutput::V2".to_string(),
+            });
+    };
```

Comment:
No that's wrong; you still need to support dWallets that were created when we were in v1, which will have v1 outputs, and in that case they will use the `targeted` dkg/presign option, and they would have to have `Some()` dkg output